pkg/backends/newrelic/newrelic.go

```diff
@@ -0,0 +1,441 @@
+package newrelic
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"runtime"
+	"sync/atomic"
+	"time"
+
+	"github.com/atlassian/gostatsd"
+	stats "github.com/atlassian/gostatsd/pkg/statser"
+
+	"github.com/cenkalti/backoff"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const (
+	// BackendName is the name of this backend.
+	BackendName                  = "newrelic"
+	integrationName              = "com.newrelic.gostatsd"
+	integrationVersion           = "2.0.0"
+	protocolVersion              = "2"
+	defaultUserAgent             = "gostatsd"
+	defaultMaxRequestElapsedTime = 15 * time.Second
+	defaultClientTimeout         = 9 * time.Second
+	// defaultMetricsPerBatch is the default number of metrics to send in a single batch.
+	defaultMetricsPerBatch = 1000
+	// maxResponseSize is the maximum response size we are willing to read.
+	maxResponseSize     = 10 * 1024
+	maxConcurrentEvents = 20
+
+	defaultEnableHttp2 = false
+)
+
+var (
+	// defaultMaxRequests is the number of parallel outgoing requests to New Relic.  As this mixes both
+	// CPU (JSON encoding, TLS) and network bound operations, balancing may require some experimentation.
+	defaultMaxRequests = uint(2 * runtime.NumCPU())
+)
+
+// Client represents a New Relic client.
+type Client struct {
+	address      string
+	eventType    string
+	flushType    string
+	metricPrefix string
+	tagPrefix    string
+	//Options to define your own field names to support other StatsD implementations
```

Comment:
Removed metric prefix, but retained tag prefix since that is commonly asked for. 
Usecase being to simply add "tag_" to avoid the potential of other attributes overlapping, due to the flat kv pair structure.